### DISCUSSION TITLE
Using a different type instead of tonic::Status

### DISCUSSION THREAD

**Discussion by k-bx** [hyperium/tonic#1481] [Q&A]
Is there a way to use an error type other than `tonic::Status` as an error? My app defines its own type `AppError`, implementing all the `From` instances that allow me to use the `?` operator. It worked well with Actix-Web framework. In Tonic, however, every call that returns an Err will get this additional construct:

```rust
        .map_err(Into::<AppError>::into)?;
```

Is there a way to avoid this?

**Comment by LucioFranco** [comment]
The trick is ti implement `From<Error> for Status` and using the `?` to convert it.

**Reply by k-bx** [reply]
@LucioFranco sorry if I mis-understand this one. For example, my function returns an `Err<_, sea_orm::DbErr>` type. If I'll try to implement this:

```
impl From<sea_orm::DbErr> for tonic::Status {
    fn from(e: sea_orm::DbErr) -> Self {
        eprintln!("DbErr: {:?}", e);
        tonic::Status::internal("Internal Server Error")
    }
}
```

I'll get an error:

```
203 | impl From<sea_orm::DbErr> for tonic::Status {
    | ^^^^^--------------------^^^^^-------------
    | |    |                        |
    | |    |                        `tonic::Status` is not defined in the current crate
    | |    `sea_orm::DbErr` is not defined in the current crate
    | impl doesn't use only types from inside the current crate
    |
    = note: define and implement a trait or new type instead
```

Or did you mean something else?

**Reply by k-bx** [reply]
@LucioFranco any info on this one?
